### PR TITLE
HEAT-179 expose active agencies on Developer portal.

### PR DESCRIPTION
**Pull request by williamfalconeruk**
Added a number of things:

- Readme corrections.
- Strapi Schema update with manual fixes.
- Make nunjucks template logic based on different states of active agencies. 
- Unit tests added for `environment/` route and associated page logic.
- Docker Compose minor enhancements for .env file.
- tweaked the lint-fix to match CI options.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -20,6 +20,10 @@ services:
       - "3000:3000"
     environment:
       - REDIS_HOST=redis-dev-portal
-
+      - PRODUCT_ID=${PRODUCT_ID}
```
Comment (williamfalconeruk) [inline docker-compose.yml:23]: Added these so i could run node container from local .env file

File: server/views/pages/environment.njk [hunk 2]
```diff
@@ -17,37 +17,47 @@
...
-      <td>{% if component.environment.health_path|length %}<a href="{{ component.environment.url }}{{ component.environment.health_path }}" target="_blank">{{ component.environment.health_path }}</a> (opens in new window){% endif %}</td>
+      <td>{% if component.environment.health_path|length %}<a data-test="health" href="{{ component.environment.url }}{{ component.environment.health_path }}" target="_blank">{{ component.environment.health_path }}</a> (opens in new window){% endif %}</td>
     </tr>
     <tr>
       <th>Cluster</th>
-      <td>{{ component.environment.cluster }}</td>
+      <td data-test="cluster">{{ component.environment.cluster }}</td>
+    </tr>
+    <tr>
+      <th>Active Agencies</th>
```
Comment (williamfalconeruk) [inline server/views/pages/environment.njk:53]: Made the active agencies more user readable so end users can identify:
- information not provided on active agencies
- when the all agencies flag (`***`) has been used
- when there's a list of agency codes. 
currently we do not the agency codes translate to descriptive names - one for a later ticket, pending identifying a place to get these that's up to date.

File: README.md [hunk 3]
```diff
@@ -22,7 +22,7 @@ The app requires:
 
 The TypeScript types for Strapi are imported via the Open API (Swagger) docs.
 
-This are stored in [`./server/@types/`](./server/@types/). Unfortunately strapi doesn't provide access in the normal way and the documentaion file must first be downloaded from the running service catalogue i.e. `kubectl -n dps-toolkit cp hmpps-service-catalogue-6778c7667f-qwmj5:/opt/app/src/extensions/documentation/documentation/1.0.0/full_documentation.json  ./full_documentation.json`
+This are stored in [`./server/@types/`](./server/@types/). Unfortunately strapi doesn't provide access in the normal way and the documentaion file must first be downloaded from the running service catalogue i.e. `kubectl -n hmpps-portfolio-management-dev cp hmpps-service-catalogue-6778c7667f-qwmj5:/opt/app/src/extensions/documentation/documentation/1.0.0/full_documentation.json  ./full_documentation.json`
```
Comment (williamfalconeruk) [inline README.md:25]: Noticed this didn't align with the current environment.

File: package.json [hunk 4]
```diff
@@ -19,7 +19,7 @@
     "watch-node-feature": "export $(cat feature.env) && nodemon --watch dist/ $NODE_DEBUG_OPTION dist/server.js | bunyan -o short",
     "start-feature:dev": "concurrently -k -p \"[{name}]\" -n \"Views,TypeScript,Node,Sass\" -c \"yellow.bold,cyan.bold,green.bold,blue.bold\" \"npm run watch-views\" \"npm run watch-ts\" \"npm run watch-node-feature\" \"npm run watch-sass\"",
     "lint": "eslint . --cache --max-warnings 0",
-    "lint-fix": "eslint . --fix",
+    "lint-fix": "eslint .  --cache --max-warnings 0 --fix",
```
[the remark this answers is not in the document]
Comment (psoleckimoj) [reply package.json:22]: Probably worth pushing back into the typescript template too? I think I added this from another project.

[the remark this answers is not in the document]
Comment (williamfalconeruk) [reply package.json:22]: I'll have a chat about this with you, as not worked on that project before.

File: server/views/pages/environment.njk (hunk 2 above)
Comment (psoleckimoj) [reply server/views/pages/environment.njk:53]: Other option is moving this logic into the route/code so there's less code in the templates.

Comment (psoleckimoj) [reply server/views/pages/environment.njk:53]: Happy to LGTM it but worth moving this into a layer above the template.

**Review by psoleckimoj — APPROVED** [review]
LGTM!

File: server/views/pages/environment.njk (hunk 2 above)
Comment (williamfalconeruk) [reply server/views/pages/environment.njk:53]: i'll take a look in my next pass!

**Pull request merged**